src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayOverviewOrchestrator.cs

```diff
@@ -40,8 +39,131 @@ public async Task<RoatpGatewayApplicationViewModel> GetOverviewViewModel(GetAppl
+
+            var viewmodel = new RoatpGatewayApplicationViewModel(applicationData);
+            viewmodel.Sequences = GetCoreGatewayApplicationViewModel();
+
+            var savedStatuses = await _applyApiClient.GetGatewayPageAnswers(request.ApplicationId);
+            if (savedStatuses != null && !savedStatuses.Any())
+            {
+                var providerRoute = application.ApplyData.ApplyDetails.ProviderRoute;
+                await _sectionsNotRequiredService.SetupNotRequiredLinks(request.ApplicationId, request.UserName, viewmodel, providerRoute);
+            }
+            else
+            {
+                foreach (var currentStatus in savedStatuses)
+                {
+                    // Inject the statuses into viewmodel
+                    viewmodel.Sequences.SelectMany(seq => seq.Sections).FirstOrDefault(sec => sec.PageId == currentStatus.PageId).Status = currentStatus?.Status;
+                }
+            }
+
+            viewmodel.ReadyToConfirm = CheckIsItReadyToConfirm(viewmodel);
+
+            return viewmodel;
+        }
+
+        public async Task<RoatpGatewayApplicationViewModel> GetConfirmOverviewViewModel(GetApplicationOverviewRequest request)
+        {
+            var application = await _applyApiClient.GetApplication(request.ApplicationId);
+            if (application is null)
+            {
+                return null;
+            }
+
+            var applicationData = GetApplicationData(application);
+
+            var viewmodel = new RoatpGatewayApplicationViewModel(applicationData);
+            viewmodel.Sequences = GetCoreGatewayApplicationViewModel();
+
+            var savedStatuses = await _applyApiClient.GetGatewayPageAnswers(request.ApplicationId);
+            if (savedStatuses != null && !savedStatuses.Any())
+            {
+                viewmodel.ReadyToConfirm = false;
+                return viewmodel;
+            }
+            else
+            {
+                foreach (var currentStatus in savedStatuses)
+                {
+                    // Inject the statuses and comments into viewmodel
+                    viewmodel.Sequences.SelectMany(seq => seq.Sections).FirstOrDefault(sec => sec.PageId == currentStatus.PageId).Status = currentStatus?.Status;
+                    viewmodel.Sequences.SelectMany(seq => seq.Sections).FirstOrDefault(sec => sec.PageId == currentStatus.PageId).Comment = currentStatus?.Comments;
+                }
+            }
+
+            viewmodel.ReadyToConfirm = CheckIsItReadyToConfirm(viewmodel);
+
+            return viewmodel;
+        }
+
+        public void ProcessViewModelOnError(RoatpGatewayApplicationViewModel viewModelOnError, RoatpGatewayApplicationViewModel viewModel, ValidationResponse validationResponse)
+        {
+            if (validationResponse.Errors != null && validationResponse.Errors.Any())
+            {
+                viewModelOnError.IsInvalid = true;
```

Comment:
This could do with being split out into an AutoMapper class

Reply:
I disagree although I do have an irrational hatred of Automapper.